enabling other organisms

vladpetyuk opened the pull request:
may need explicitly add GOSemSim to Imports or expose getDb function (or similar function linking organism name to org.X.eg.db) in other way.   I'll add organism name conversion to make sure 'graphite' compatible with 'GOSemSim'.  E.g. homo sapiens is 'hsapiens' in 'graphite', but 'human' in 'GOSemSim'.

vladpetyuk committed:
enabling other organisms

vladpetyuk committed:
ensure organism name conversion

ensure compatibility between graphite and GOSemSim.

vladpetyuk committed:
boolean bug fix

vladpetyuk committed:
typo fix

vladpetyuk committed:
tweaking error message

GuangchuangYu commented:
The organism translation is good.

But their is a bug in ID conversion. Not all annotation db is org.X.eg.db, for example yeast is org.Sc.sgd.db. see https://support.bioconductor.org/p/64729/.

The original version use EXTID2NAME from DOSE do import DB via GOSemSim::getDb and more compatible. Can you modify this part?

vladpetyuk committed:
switch entrez to name procedure

vladpetyuk committed:
safer pathName parsing

string split by ": " is dangerous because sometimes colon is in the pathway names.  For example (5752278):
"Caenorhabditis elegans: Antigen processing: Ubiquitination & Proteasome degradation"
